File: AdaptiveTD/AdaptiveTD/AdaptiveTD/GUIButton.cs
using System;
using System.Collections.Generic;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework.Input;

namespace AdaptiveTD
{
    class GUIButton
    {
        Texture2D texture;
        Vector2 position;
        Color color;
        Keys keyBinding;
        public Keys KeyBinding
        {
            get { return keyBinding; }
        }
        public Color Color
        {
            set { color = value; }
        }

        public GUIButton(Texture2D texture, Vector2 position, Keys keyBinding)
        {
            this.texture = texture;
            this.position = position;
            color = Color.White;
            this.keyBinding = keyBinding;
        }

        public bool ButtonClicked(float x, float y)
        {
            if (x >= position.X && x <= position.X + texture.Width && y >= position.Y && y <= position.Y + texture.Height)
                return true;
            else
                return false;
        }

        public void Draw(SpriteBatch spriteBatch)
        {
            spriteBatch.Draw(texture, position, color);
        }
    }
}

File: LibGDXProjects/AdaptiveTD/src/no/uia/adaptiveTD/GUIButton.java
package no.uia.adaptiveTD;

import com.badlogic.gdx.Input.Keys;
import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.Sprite;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;
import com.badlogic.gdx.math.Vector2;

public class GUIButton {

	Sprite sprite;
	Vector2 position;
	
	Color color;
	public Color getColor() {
		return color;
	}
	public void setColor(Color color) {
		this.color = color;
	}
	Keys keybinding;
	public Keys getKeybinding() {
		return keybinding;
	}
	
	public GUIButton(Sprite sprite, Vector2 position, Keys keybinding)
	{
		this.sprite = sprite;
		this.position = position;
		color = Color.WHITE;
		this.keybinding = keybinding;
	}
	
	public boolean buttonClicked(float x, float y)
	{
        if (x >= position.x && x <= position.x + sprite.getWidth() && y >= position.y && y <= position.y + sprite.getHeight())
        	return true;
        else
        	return false;
	}
	
	public void draw(SpriteBatch spriteBatch)
	{
		spriteBatch.draw(sprite, position.x, position.y);
	}
}
